opensrp-app/src/main/java/org/smartregister/repository/TaskRepository.java

```diff
@@ -112,6 +121,17 @@ public static void createTable(SQLiteDatabase database) {
         database.execSQL(CREATE_TASK_PLAN_GROUP_INDEX);
     }
 
+    /**
+     * Migrate the older database by add restriction columns and changing of priority to enum
+     * @param database the database being upgraded
+     */
+    public static void updatePriorityToEnumAndAddRestrictions(@NonNull SQLiteDatabase database) {
+        DatabaseMigrationUtils.addColumnIfNotExists(database, TASK_TABLE, RESTRICTION_REPEAT, VARCHAR);
+        DatabaseMigrationUtils.addColumnIfNotExists(database, TASK_TABLE, RESTRICTION_START, VARCHAR);
+        DatabaseMigrationUtils.addColumnIfNotExists(database, TASK_TABLE, RESTRICTION_END, VARCHAR);
+        DatabaseMigrationUtils.recreateSyncTableWithExistingColumnsOnly(database, TASK_TABLE, COLUMNS, CREATE_TASK_TABLE);
+        database.execSQL(String.format("UPDATE %s SET %s=? WHERE %s =? ", TASK_TABLE, PRIORITY, PRIORITY), new Object[]{"3", "3"});
```

Comment:
1. I think  this query is wrong
2. Have you tested this out and line 132?

Reply:
Yes tested and now it works